docs/storage-surface.rst

```diff
@@ -0,0 +1,673 @@
+:func:`lookup_bucket <gcloud.storage.api.lookup_bucket>`
+
+.. code-block:: python
+
+   >>> non_existent = storage.lookup_bucket(bucket_name)
+   >>> print non_existent
+   None
+
+To retrieve multiple buckets in a single request
+
+.. code-block:: python
+
+   >>> bucket1, bucket2, bucket3 = storage.get_buckets('bucket-name1',
+   ...                                                 'bucket-name2',
+   ...                                                 'bucket-name3')
+
+This is equivalent to
+
+.. code-block:: python
+
+   >>> with storage.Batch():
+   ...     bucket1 = storage.get_bucket('bucket-name1')
+   ...     bucket2 = storage.get_bucket('bucket-name2')
+   ...     bucket3 = storage.get_bucket('bucket-name3')
+
+To list all buckets associated to the default project
+
+.. code-block:: python
+
+   >>> for bucket in storage.list_buckets():
+   ...     print bucket
+   <Bucket: foo>
+   <Bucket: bar>
+   <Bucket: baz>
+
+or to use a project other than the default
+
+.. code-block:: python
+
+   >>> for bucket in storage.list_buckets('some-project'):
+   ...     print bucket
+
+To limit the list of buckets returned,
+:func:`list_buckets() <gcloud.storage.list_buckets>` accepts optional
+arguments
+
+.. code-block:: python
+
+   >>> bucket_iterator = storage.list_buckets(max_results=2,
+   ...                                        page_token='next-bucket-name',
+   ...                                        prefix='foo',
+   ...                                        projection='noAcl',
+   ...                                        fields=None)
+   >>> for bucket in bucket_iterator:
+   ...     print bucket
+
+See the `buckets list`_ documentation for details.
+
+.. _buckets list: https://cloud.google.com/storage/docs/json_api/v1/buckets/list
+
+To delete a bucket
+
+.. code-block:: python
+
+   >>> storage.delete_bucket(bucket_name)
+
+.. warning::
+  Deleting a bucket should happen very infrequently. Be careful that you
+  actually mean to delete the bucket.
+
+.. note::
+  We use the term blob interchangeably with "object" when referring to the
+  API. The Google Cloud Storage documentation use object, but we use ``blob``
+  instead to avoid confusion with the Python builtin ``object``.
+
+In the case that the bucket has existing blobs, the backend
+will return a `409 conflict`_ and raise
+
+.. code-block:: python
+
+   >>> storage.delete_bucket(bucket_name)
+   Traceback (most recent call last):
+     File "<stdin>", line 1, in <module>
+   gcloud.exceptions.Conflict: 409 Some Message
+
+this can be addressed by using the ``force`` keyword
+
+   >>> storage.delete_bucket(bucket_name, force=True)
+
+Even using ``force=True`` will fail if the bucket contains more than 256
+blobs. In this case, the blobs should be deleted manually first.
+
+.. _409 conflict: http://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_Error
+
+Working with Buckets
+--------------------
+
+To create a bucket object directly
```

Comment:
Ouch! makes me want to delete `storage.create_bucket()` and `storage.delete_bucket()`.